homeassistant/components/cover/brunt.py

```diff
@@ -0,0 +1,199 @@
+
+CLOSED_POSITION = 0
+OPEN_POSITION = 100
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+POSITION_MAP = {
+    0: STATE_CLOSED,
+    100: STATE_OPEN,
+}
+MOVE_STATES_MAP = {
+    1: STATE_OPENING,
+    2: STATE_CLOSING,
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the brunt platform."""
+    # pylint: disable=no-name-in-module
+    from brunt import BruntAPI
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    bapi = BruntAPI(username=username, password=password)
+    try:
+        things = bapi.getThings()['things']
+        if not things:
+            _LOGGER.error("No things present in account.")
+        else:
+            add_devices(BruntDevice(
+                hass, bapi, thing['NAME'],
+                thing['thingUri']) for thing in things)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        _LOGGER.error("%s", ex)
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+
+
+class BruntDevice(CoverDevice):
+    """
+    Representation of a Brunt cover device.
+
+    Contains the common logic for all Brunt devices.
+    """
+
+    def __init__(self, hass, bapi, name, thing_uri):
+        """Init the Brunt device."""
+        self._bapi = bapi
+        self._name = name
+        self._thing_uri = thing_uri
+
+        self._state = STATE_UNKNOWN
+        self._available = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device as reported by tellcore."""
+        return self._name
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._available
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return int(self._state['currentPosition'])
+
+    @property
+    def request_cover_position(self):
+        """
+        Return requested position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return int(self._state['requestPosition'])
+
+    @property
+    def move_state(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 when stopped, 1 when opening, 2 when closing
+        """
+        return int(self._state['moveState'])
+
+    @property
+    def device_state_attributes(self):
+        """Return the detailed device state attributes."""
+        return {
+            ATTR_CURRENT_POSITION: self.current_cover_position,
+            ATTR_REQUEST_POSITION: self.request_cover_position
```

Comment:
REQUESTED ?

Reply:
@dgomes, that's the name used in the response from the endpoint, so I decided to keep that in sync. So I will make a change to the name of that in the interface (the content of ATTR_REQUEST_POSITION), but I don't want to change the code to maintain traceability from endpoints, through sdk, to here in a readible way. That is the current look, will update now.
![image](https://user-images.githubusercontent.com/13749212/42132122-0bfba5b6-7d12-11e8-83a3-8ea21caaefd2.png)

Reply:
and the new look:
![image](https://user-images.githubusercontent.com/13749212/42132139-63cd7490-7d12-11e8-81b5-5850fa45b579.png)

Reply:
Sorry, I was mislead by your function documentation that stated "Return requested position of cover."

Just looking for overall consistency, if it makes sense to be "request", please change it back.